core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -137,7 +138,7 @@ abstract class AbstractFetcherThread(name: String,
                     // Once we hand off the partition data to the subclass, we can't mess with it any more in this thread
                     processPartitionData(topicAndPartition, currentPartitionFetchState.offset, partitionData)
                   } catch {
-                    case ime: InvalidMessageException =>
+                    case ime: CorruptRecordException =>
```

Comment:
This is the one area where some server code and user code share exceptions that needed to be updated. This is the answer, to your above question (working on responding with detail there) about changing the inheritance for InvalidMessageException. Now any user catching InvalidMessageException can still catch that and the new CorruptRecordException will also be caught.